enderio-base/src/main/java/com/enderio/base/common/item/tool/ColdFireIgniter.java

```diff
@@ -0,0 +1,118 @@
+                Level level = context.getLevel();
+                BlockPos blockpos = context.getClickedPos().relative(context.getClickedFace());
+                BlockState coldFireState = EIOBlocks.COLD_FIRE.get().getStateForPlacement(new BlockPlaceContext(context));
+                if (coldFireState.canSurvive(context.getLevel(), blockpos)) {
+                    level.playSound(player, blockpos, SoundEvents.FLINTANDSTEEL_USE, SoundSource.BLOCKS, 1.0F, level.getRandom().nextFloat() * 0.4F + 0.8F);
+                    level.setBlock(blockpos, coldFireState, Block.UPDATE_ALL_IMMEDIATE);
+                    level.gameEvent(player, GameEvent.BLOCK_PLACE, blockpos);
+                    if (player instanceof ServerPlayer serverPlayer) {
+                        CriteriaTriggers.PLACED_BLOCK.trigger(serverPlayer, blockpos, itemstack);
+                    }
+                    return InteractionResult.sidedSuccess(level.isClientSide());
+                }
+            }
+            return InteractionResult.FAIL;
+        }
+        LogManager.getLogger().warn("could not find FluidCapability on ColdFireIgniter");
+        return super.useOn(context);
+    }
+
+    @Override
+    public boolean isBarVisible(ItemStack stack) {
+        return getTankCap(stack).isPresent();
+    }
+
+    @Override
+    public int getBarColor(ItemStack stack) {
+        return 0x99BD42;
+    }
+
+    @Override
+    public int getBarWidth(ItemStack stack) {
+        Optional<IFluidHandlerItem> tankCap = getTankCap(stack);
+        if (tankCap.isPresent()) {
+            IFluidHandlerItem fluidHandler = tankCap.get();
+            return Math.round(fluidHandler.getFluidInTank(0).getAmount() * 13f / fluidHandler.getTankCapacity(0));
+        }
+        return super.getBarWidth(stack);
+    }
+
+
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> components, TooltipFlag flag) {
+        super.appendHoverText(stack, level, components, flag);
+        getTankCap(stack).ifPresent(handler -> {
+            boolean isOneTank = handler.getTanks() == 1;
+            if (!isOneTank) {
+                components.add(new TextComponent("Fluids:"));
+            }
+            for (int i = 0; i < handler.getTanks(); i++) {
+                String prefix = isOneTank ? "" : i + ": ";
+                Component postFix = handler.getFluidInTank(i).isEmpty() ? new TextComponent("") : handler.getFluidInTank(i).getDisplayName();
+                components.add(new TextComponent(prefix + handler.getFluidInTank(i).getAmount() + " / " + handler.getTankCapacity(i) + " ").append(postFix));
+            }
+        });
+    }
+    @Nullable
+    @Override
+    public MultiCapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundTag nbt, MultiCapabilityProvider provider) {
+        provider.addSimple(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY,
+            new AcceptingFluidItemHandler(stack, 1000, Fluids.LAVA)
```

Comment:
Is it possible to supply this with a data pack? for packs that do not wish to have vapor of levity, or wish to use their own custom fluid?

Reply:
The fluid handler is built to also accept tags, so adding a coldfire_fluid tag is possible

Reply:
I'd say this would be better served with a config option?

Reply:
configs are imo inferior to tags, if it's possible to do with a tag. Other mods can't change our config, but our tags. Users can change both. A config is just a bit more visible to the user, but if we have an extensive wiki describing which tag controls what and having a note in our configs pointing to the wiki we should have the same visibility to users, but a much higher customizability